### PR TITLE
Development

### PR DESCRIPTION
**Pull request by nandras88**
All the sprints are completed.

http://jira.greenfox.academy/secure/RapidBoard.jspa?rapidView=38&projectKey=FLU&view=detail&selectedIssue=FLU-4

### CODE REVIEW

File: sf_car/src/main.c [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+	}
+}
+static void MPU_Config(void)
+{
+	MPU_Region_InitTypeDef MPU_InitStruct;
+
+	HAL_MPU_Disable();
+
+	MPU_InitStruct.Enable = MPU_REGION_ENABLE;
+	MPU_InitStruct.BaseAddress = 0x20010000;
```
Comment (leventekobor94) [inline sf_car/src/main.c]: @nandras88 why this value?

**Review by samsmith94 — CHANGES_REQUESTED** [review]
- motor_forward_setup() and motor_backward_setup() funcitions should be inside 1 function, and find a better name for it. The code inside these functions doesn't refer to its usage. They are just gpio settings and these 2 functions are not called via touch screen buttons
- unneccessary new lines should be removed

File: sf_car/src/main.c [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+#include "stm32746g_discovery_ts.h"
+#include <string.h>
+#include <stdlib.h>
+
+UART_HandleTypeDef uart_handle;  //uart typedef
+TIM_HandleTypeDef servo_handle;  //pwm timer typedef
+TIM_OC_InitTypeDef servo_Config; // pwm typedef for servo
+GPIO_InitTypeDef servo_motor;	//servo_motor typedef
+TS_StateTypeDef ts_state;		 // steering for button
+GPIO_InitTypeDef motor_forward;  //engine turning forward
```
Comment (iringo) [inline sf_car/src/main.c:13]: You meant here engine moving and not turning forward? Same in line 17?

File: sf_car/src/main.c [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+TIM_OC_InitTypeDef motor_Config; //pwm typedef for motor
+TIM_HandleTypeDef motor_handle;  //pwm timer typedef for motor
+GPIO_InitTypeDef motor_pwm;		 //pwm motor gpio init
+GPIO_InitTypeDef motor_backward; //engine turning backward
+
+volatile int x;
```
Comment (iringo) [inline sf_car/src/main.c:19]: Maybe a more self-explanatory varibale name would be better here, like: x_coordinate_for_LCD or something similar.

File: sf_car/src/main.c [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+}
+
+void servo_pwm_setup()
+{
+
+	__HAL_RCC_TIM3_CLK_ENABLE();
+
+	//pwm setup
+	servo_handle.Instance = TIM3;
+	servo_handle.Init.Period = 2000;
```
Comment (iringo) [inline sf_car/src/main.c:66]: Why exactly these concrete numbers? What is the logic behind it?

File: sf_car/src/main.c [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+	servo_handle.Init.Prescaler = 1080;
+	servo_handle.Init.ClockDivision = TIM_CLOCKDIVISION_DIV1;
+	servo_handle.Init.CounterMode = TIM_COUNTERMODE_UP;
+	HAL_TIM_Base_Init(&servo_handle);
+	HAL_TIM_Base_Start(&servo_handle);
+
+	//pwm config
+	servo_Config.OCMode = TIM_OCMODE_PWM1;
+	servo_Config.Pulse = 145;
```
Comment (iringo) [inline sf_car/src/main.c:75]: A very similar question why exactly 145 for the pulse?

File: sf_car/src/main.c [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+	HAL_GPIO_Init(GPIOA, &motor_forward);
+}
+
+void motor_backward_setup()
+{
+
+	//motor backward
+	__HAL_RCC_GPIOF_CLK_ENABLE();
+
+	motor_backward.Pin = GPIO_PIN_10;
```
Comment (iringo) [inline sf_car/src/main.c:102]: Why PIN10? Is this related to the motor's documentation? Same question for line 115!

File: sf_car/src/main.c [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+}
+
+void motor_pwm_setup()
+{
+
+	__HAL_RCC_TIM2_CLK_ENABLE();
+
+	//motor pwm setup
+	motor_handle.Instance = TIM2;
+	motor_handle.Init.Period = 1000;
```
Comment (iringo) [inline sf_car/src/main.c:130]: Once again a similar question: why these exact numbers for the period and prescale?

File: sf_car/src/main.c [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+	servo_motor_setup();
+	servo_pwm_setup();
+	lcd_setup();
+	gui_interface();
+	motor_forward_setup();
+	motor_backward_setup();
+	motor_pin_setup();
+	motor_pwm_setup();
+
+	HAL_GPIO_WritePin(GPIOF, GPIO_PIN_10, GPIO_PIN_SET);
```
Comment (iringo) [inline sf_car/src/main.c:202]: What function does this WritePin function serve?

File: sf_car/src/main.c [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+			{
+				TIM3->CCR1 = 150;
+			}
+			else if (x >= 380 && x <= 455 && y >= 150 && y <= 225)
+			{
+				TIM3->CCR1 = 130;
+			}
+			else if (x >= 380 && x <= 455 && y >= 10 && y <= 85)
+			{
+				HAL_GPIO_TogglePin(GPIOA, GPIO_PIN_0);
```
Comment (iringo) [inline sf_car/src/main.c:235]: What is the purpose of these TogglePin functions?

**Pull request merged**